enricher/api/src/main/java/io/fabric8/maven/enricher/api/AbstractHealthCheckEnricher.java

```diff
@@ -0,0 +1,90 @@
+package io.fabric8.maven.enricher.api;
+
+import java.util.List;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.Probe;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+
+/**
+ * Enriches containers with health check probes.
+ */
+public abstract class AbstractHealthCheckEnricher extends BaseEnricher {
+
+    public AbstractHealthCheckEnricher(EnricherContext buildContext, String name) {
+        super(buildContext, name);
+    }
+
+    @Override
+    public void addMissingResources(KubernetesListBuilder builder) {
+
+        List<HasMetadata> items = builder.getItems();
+        for (HasMetadata item : items) {
+            if (item instanceof Deployment) {
+                Deployment deployment = (Deployment) item;
+                Container container = getContainer(deployment);
+                if (container != null) {
+                    if (container.getReadinessProbe() == null) {
+                        Probe probe = getReadinessProbe();
+                        if (probe != null) {
+                            log.info("Adding readiness " + describe(probe));
+                            container.setReadinessProbe(probe);
+                        }
+                    }
+
+                    if (container.getLivenessProbe() == null) {
+                        Probe probe = getLivenessProbe();
+                        if (probe != null) {
+                            log.info("Adding liveness " + describe(probe));
+                            container.setLivenessProbe(probe);
+                        }
+                    }
+                }
+            }
+        }
+        builder.withItems(items);
+    }
+
+    private Container getContainer(Deployment deployment) {
+        if (deployment.getSpec() != null &&
+                deployment.getSpec().getTemplate() != null &&
+                deployment.getSpec().getTemplate().getSpec() != null &&
+                deployment.getSpec().getTemplate().getSpec().getContainers() != null &&
+                deployment.getSpec().getTemplate().getSpec().getContainers().size() > 0) {
+
+            List<Container> containers = deployment.getSpec().getTemplate().getSpec().getContainers();
+            return containers.get(containers.size() - 1); // last one
```

Comment:
In deed, f-m-p is supposed to deal with multiple containers from multiples images within a pod. However, currently the 'generators' (which are supposed to create default images) kick in by default only if there is no other image configured. 

Maybe it would make sense to establish some sort of 'communication' between generators and enrichers, eg. by setting a certain marker in the image configuration.

Then the algo could be:

* If there is only one image / container in the deployment config and we are a spring-boot project, lets add the health check.
* If there are multiple containers, lets check for this 'marker' and add health checks to each of the marked containers.

The question of course is, what could be this "marker" ? A label or annotation is probably the easiest thing, however the question how this can be added to a `Deployment` config. Maybe by the enricher which adds this `Deployment` ? (if done manually, one need to add this annotation oneself in the configuration).

Still in brainstorming mode ;-) ...